src/puppetlabs/puppetdb/mq_listener.clj

```diff
@@ -137,22 +138,12 @@
 ;; architecture.
 ;;
 
-(defn wrap-with-meter
-  "Wraps a message processor `f` and a `meter` such that `meter` will be marked
-  for each invocation of `f`."
-  [f meter]
-  (fn [msg]
-    (mark! meter)
-    (f msg)))
-
-(defn try-parse-command
-  "Tries to parse `msg`, returning the parsed message if the parse is
-  successful or a Throwable object if one is thrown."
+(defn parse-command
   [msg]
   (try+
    (cmd/parse-command msg)
    (catch Throwable e
-     e)))
+     (throw+ {:kind ::parse-error} e "Error parsing command"))))
```

Comment:
Assuming this exception is just for the narrow use of message-handler-with-retries, I might consider moving this function down to just before that function.